xarray/backends/api.py

```diff
@@ -1827,18 +1842,242 @@ def to_zarr(
                     "mode='r+'. To allow writing new variables, set mode='a'."
                 )
 
+    return write_to_zarr_store(
+        dataset,
+        zstore,
+        encoding,
+        compute,
+        chunkmanager_store_kwargs,
+        encode_coordinates=True,
+    )
+
+
+def write_to_zarr_store(
+    dataset, store, encoding, compute, chunkmanager_store_kwargs, encode_coordinates
+):
     writer = ArrayWriter()
     # TODO: figure out how to properly handle unlimited_dims
-    dump_to_store(dataset, zstore, writer, encoding=encoding)
+    dump_to_store(
+        dataset, store, writer, encoding=encoding, encode_coordinates=encode_coordinates
+    )
     writes = writer.sync(
         compute=compute, chunkmanager_store_kwargs=chunkmanager_store_kwargs
     )
 
     if compute:
-        _finalize_store(writes, zstore)
+        _finalize_store(writes, store)
     else:
         import dask
 
-        return dask.delayed(_finalize_store)(writes, zstore)
+        return dask.delayed(_finalize_store)(writes, store)
+    return store
+
+
+def initialize_zarr(
+    ds: Dataset,
+    store: MutableMapping,
+    *,
+    region_dims: Iterable[Hashable] = tuple(),
+    mode: Literal["w", "w-"] = "w-",
+    zarr_version: int | None = None,
+    consolidated: bool | None = None,
+    encoding: dict | None = None,
+    **kwargs,
+) -> Dataset:
+    """
+    Initialize a Zarr store with metadata.
+
+    This function initializes a Zarr store with all indexed coordinate variables, and
+    metadata for every variable in the dataset.
+    If ``region_dims`` is specified, it will also
```

Comment:
Not a necessary change, but one plausible way to frame this is
- default `region_dims=[...]`
- ...and then there's no need for the `if` — we default to writing the minimum and returning the maximum.

Reply:
Interesting suggestion!

Reply:
There's an edge case around scalars. We would write them if the user wants region writes, but not write them if they don't. So we need to distinguish between `region_dims=()`, `region_dims=(...,)` and `region_dims=tuple(ds.dims)`. If those last two are identical then we can't handle the scalar case.

Reply:
OK interesting point.

OOI, is this function helpful if we're not doing region writes?

Reply:
Yeah, on reflection — is there a case where we really don't want to write scalars? (genuine question, very possibly there is...)